lib/logstash/filters/mutate.rb

```diff
@@ -35,6 +35,17 @@ class LogStash::Filters::Mutate < LogStash::Filters::Base
   # filters.
   config :remove, :validate => :array, :deprecated => true
 
+  # Keep only the specified fields.
+  #
+  # Example:
+  # [source,ruby]
+  #     filter {
+  #       mutate {
+  #         keep => [ "message", "subject" ]  # Removes all the fields except message and subject
+  #       }
+  #     }
+  config :keep, :validate => :array
```

Comment:
what do you think of naming this `select`, since we choose to only select these specified fields. Keep is misleading to me a bit because we keep all fields by default.

I can go both ways on this though, not a big deal since we have documentation about what it does.